src/test/java/ch/jalu/collectionbehavior/SetTest.java

```diff
@@ -1,333 +1,499 @@
+            if (skipsWrappingForOwnClass) {
+                return Stream.of(testLogic.skipsWrappingForOwnClass());
+            }
+            return Stream.empty();
+        }
+
+        private Stream<DynamicTest> createTestForDuplicatesOnCreation() {
+            if (!setCreator.canEncounterDuplicateArguments()) {
+                Preconditions.checkState(acceptsDuplicatesOnCreation);
+                return Stream.empty();
+            }
+
+            return acceptsDuplicatesOnCreation
+                ? Stream.of(testLogic.acceptsDuplicatesOnCreation(elementOrder))
+                : Stream.of(testLogic.rejectsDuplicatesOnCreation());
+        }
     }
 
-    private static void assertContainsButNotInOrder(Set<Integer> set, Integer... values) {
-        assertThat(set, containsInAnyOrder(values));
-        assertThat(set, not(contains(values)));
-        // Of course it's still possible that by coincidence the Set has the same order, but we just have to live with this
+    private static final class TestLogic {
+
+        private final SetCreator setCreator;
+
+        private TestLogic(SetCreator setCreator) {
+            this.setCreator = setCreator;
+        }
+
+        DynamicTest supportsNullElements() {
+            return dynamicTest("supportsNullElements",
+                () -> assertDoesNotThrow(setCreator::createSetWithNull));
+        }
+
+        DynamicTest mayNotContainNull() {
+            return dynamicTest("mayNotContainNull",
+                () -> assertThrows(NullPointerException.class, setCreator::createSetWithNull));
+        }
+
+        DynamicTest supportsNullMethodArgs() {
+            return dynamicTest("supportsNullMethodArgs",
+                () -> verifySupportsNullArgInMethods(setCreator.createSet()));
+        }
+
+        DynamicTest rejectsNullMethodArgs() {
+            return dynamicTest("rejectsNullMethodArgs",
+                () -> verifyRejectsNullArgInMethods(setCreator.createSet()));
+        }
+
+        DynamicTest skipsWrappingForOwnClass() {
+            FromCollectionSetCreator setCopyCreator = (FromCollectionSetCreator) setCreator;
+            return dynamicTest("skipsWrappingForOwnClass", () -> {
+                Set<String> set1 = setCopyCreator.newSet(Set.of("a", "b"));
+                Set<String> set2 = setCopyCreator.newSet(set1);
+                assertThat(set1, sameInstance(set2));
+            });
+        }
+
+        DynamicTest isSequencedSet() {
+            return dynamicTest("isSequencedSet",
+                () -> assertThat(setCreator.createSet(), instanceOf(SequencedSet.class)));
+        }
+
+        DynamicTest isNotSequencedSet() {
+            return dynamicTest("isNotSequencedSet", () -> {
+                Set<String> set = setCreator.createSet();
+                assertThat(set, not(instanceOf(SequencedSet.class)));
+                assertThat(set, not(instanceOf(SequencedCollection.class)));
```

Comment:
what if the set creation is collection-based or set-based? Check that a sequencedset as input does not yield a sequecnedset, as we've seen with some list methods wrt RandomAccess